### PR TITLE
12 solution local moves

### LINKED ISSUE #12: [Solution] local_moves
Returns an iterable with all of the possible moves for the current solution state.

In our case, a move would probably be created with the [3 opt](http://tsp-basics.blogspot.com/2017/03/3-opt-move.html) logic but this is open for debate

### PR DESCRIPTION
**Pull request by AlexandreAndrade00**
closes #12 #13 #14 #17 #8 #9 #18 

based on: [Opt-3 special case](https://tsp-basics.blogspot.com/2017/03/3-opt-iterative-general-idea.html)

### CODE REVIEW

File: travelling_salesman.py [hunk 1]
```diff
@@ -102,12 +125,29 @@ def add_moves(self) -> Iterable[Component]:
         for city in self.unvisited_cities:
             yield Component(self.visited_cities[-1], city)
 
+    # https://tsp-basics.blogspot.com/2017/03/3-opt-iterative-general-idea.html
```
Comment (lutzzdias) [inline travelling_salesman.py]: I believe leaving the link in the issue/PR here is enough

**Review by lutzzdias — APPROVED** [review]
Looks good, you tested it with the sample data files right?

**Comment by AlexandreAndrade00** [comment]
> Looks good, you tested it with the sample data files right?

Yes, I tested with some samples. The results are the expected.

File: travelling_salesman.py (hunk 1 above)
Comment (AlexandreAndrade00) [reply travelling_salesman.py]: Removed

**Pull request merged**